### DISCUSSION TITLE
Company Sector Classification?

### DISCUSSION THREAD

**Discussion by webbsledge** [dgunning/edgartools#154] [General]
Does anyone know if the sector of a company is available in edgartools? It _is_ in the filings, (oddly) beside "Organization Name" in 10-K and 10-Q, but I'm having a hard time telling if edgartools is parsing this data or not.

The reason I ask if because I'm doing some investment portfolio optimization, and I need to set some sector constraints. I can do this by hand, but obviously that's less than ideal as positions change over time, and the SEC "industry" classification is a too granular.

<img width="807" alt="Screenshot 2024-11-23 at 12 52 57 PM" src="https://github.com/user-attachments/assets/e5761d8e-b37a-40cb-987a-5a12099c0939">

**Comment by dgunning** [comment]
Yes.

```python
>>> c = Company("AAPL")
>>> c.industry
'Electronic Computers'
```
<img width="1630" alt="company" src="https://github.com/user-attachments/assets/a3d1a39c-f1e1-4474-a8a6-3fa0d20988a2">

**Comment by webbsledge** [comment]
Thanks @dgunning. The XOM filing actually misled me. I thought the SEC had a parent "sector" classification into which the SICs would be characterized. After your input and further research, I believe what I'm thinking of is the MSCI/S&P Global Industry Classification Standard (GICS), which sometimes gets used by financial data providers.

**Comment by unparadise** [comment]
@dgunning Is the industry attribute removed from the most recent release?

Here is the available methods and attributes for the Company object using the dir() function.

['__abstractmethods__', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__firstlineno__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rich__', '__setattr__', '__sizeof__', '__slots__', '__static_attributes__', '__str__', '__subclasshook__', '__weakref__', '_abc_impl', '_cik', '_data', '_format_fiscal_year_date', '_get_operating_type_emoticon', 'business_address', 'cik', 'data', 'display_name', 'get_exchanges', 'get_facts', 'get_filings', 'get_financials', 'get_icon', 'get_quarterly_financials', 'get_ticker', 'is_company', 'is_individual', 'latest', 'latest_tenk', 'latest_tenq', 'mailing_address', 'name', 'not_found', 'tickers']

But when I typed the following code in IDLE, the industry data is correctly returned.

```
from edgar import *
set_identity("identity_info")
c = Company("AAPL")
c.industry

---Output---
'Electronic Computers'
```

Thank you for your help!

**Reply by unparadise** [reply]
@dgunning I did more tests and find the issue quite strange.

The same program I listed above runs correctly in IDLE but when I tried to run it with "python3 <file_name>", it returns the the following error.

```
 c.industry
AttributeError: 'Company' object has no attribute 'industry'
```

**Comment by dgunning** [comment]
Sorry for just getting back. I've had an extremely busy week away from edgartools
After the 4.0.0  release I noticed a few explicit attributes were missing - including `industry` and I have some changes in my workspace to add them back. The difference this time is that it would be `get_industry` in place of `industry`, in an attempt to make the api more  intuitive and consistent. That being said - intuitive is also subjective so users' should give feedback on the api.

Regardless, I will try to do a release today with a patch for the missing features. I will be more free after mid week to put more work into the library
